From amit.salecha@qlogic.com  Thu Jun 23 22:59:42 2011
From: Amit Kumar Salecha <amit.salecha@qlogic.com>
Date: Tue, 2 Feb 2010 04:16:20 +0000
Subject: [PATCH 05/56] netxen: fix tx timeout recovery for NX2031 chip
Git-commit: 41f3e0b1f524c47908e944a7e55f11c371ddddec
Patch-mainline: v2.6.33-rc8
References: bnc#698269, FATE#311464

For NX2031, first try to scrub interrupt before requesting firmware
reset. Return statement was missing after scrubbbing interrupt.

Signed-off-by: Vernon Mauery <vernux@us.ibm.com>
Signed-off-by: Amit Kumar Salecha <amit.salecha@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/netxen/netxen_nic_main.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/net/netxen/netxen_nic_main.c b/drivers/net/netxen/netxen_nic_main.c
index 076f826..25990ee 100644
--- a/drivers/net/netxen/netxen_nic_main.c
+++ b/drivers/net/netxen/netxen_nic_main.c
@@ -1995,7 +1995,7 @@ static void netxen_tx_timeout_task(struct work_struct *work)
 		netif_wake_queue(adapter->netdev);
 
 		clear_bit(__NX_RESETTING, &adapter->state);
-
+		return;
 	} else {
 		clear_bit(__NX_RESETTING, &adapter->state);
 		if (!netxen_nic_reset_context(adapter)) {
-- 
1.6.3.3

